src/app/shared/shared-settings-forms/units-form/units-form.component.ts

```diff
@@ -36,6 +42,14 @@ export class UnitsFormComponent implements OnInit, OnDestroy{
         this.form = this.sharedSettingsFormsService.getUnitsForm(_facility.unitSettings);
       }
       this.facility = _facility;
+      this.filteredVisits = this.onSiteVisitIdbService.onSiteVisits.getValue()
```

Comment:
same as above, instead of getting the assessments Ids by the visit, just get them directly from the assessment service

Reply:
yes, thanks!